processor/src/lib.rs

```diff
@@ -83,21 +83,21 @@ pub struct ChipletsTrace {
 // EXECUTORS
 // ================================================================================================
 
-/// Returns an execution trace resulting from executing the provided program against the provided
-/// inputs.
+/// Returns execution output and an execution trace resulting from executing the provided program
+/// against the provided inputs.
 pub fn execute(
     program: &Program,
     inputs: &ProgramInputs,
-) -> Result<ExecutionTrace, ExecutionError> {
+) -> Result<(ProgramOutputs<Felt>, ExecutionTrace), ExecutionError> {
```

Comment:
Would it complicate things much if we still just returned `ExecutionTrace` from here and modified the `ExecutionTrace` struct to contain (and return via a method) `ProgramOutputs`? If it doesn't complicate things too much, I'd prefer this approach.